src/devices/Tlc1543/Tlc1543.cs

```diff
@@ -0,0 +1,221 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Device;
+using System.Device.Gpio;
+using System.Collections.Generic;
+
+namespace Iot.Device.Tlc1543
+{
+    /// <summary>
+    /// Add documentation here
+    /// </summary>
+    public class Tlc1543 : IDisposable
+    {
+        #region Members
+
+        private readonly int _chipSelect;
+        private readonly int _dataOut;
+        private readonly int _address;
+        private readonly int _inputOutputClock;
+        private readonly int _endOfConversion;
+        private readonly bool _shouldDispose;
+        private readonly TimeSpan _conversionTime = new TimeSpan(210);
+        private GpioController _controller;
+        private Channel _chargeChannel = Channel.SelfTest512;
+        private List<int> _values = new List<int>(16);
+
+        /// <summary>
+        /// Channel used between readings to charge up capacitors in ADC
+        /// <remarks>
+        /// <br>Refer to table 2 and 3 in TLC1543 documentation for more information</br>
+        /// </remarks>
+        /// </summary>
+        public Channel ChargeChannel
+        {
+            get => _chargeChannel;
+            set
+            {
+                if (IsValidChannel((int)value))
+                {
+                    _chargeChannel = value;
+                }
+                else
+                {
+                    throw new ArgumentOutOfRangeException("ChargeChannel", "Out of range. \nMust be non-negative and less than 14. \nRefer to table 2 and 3 in TLC1543 documentation for more information");
+                }
+            }
+        }
+
+        #endregion
+
+        #region Constructors
+
+        /// <summary>
+        /// Constructor for Tlc1543
+        /// </summary>
+        /// <param name="address">Address pin</param>
+        /// <param name="chipSelect">Chip Select pin</param>
+        /// <param name="dataOut">Data Out pin</param>
+        /// <param name="inputOutputClock">I/O Clock pin</param>
+        /// <param name="endOfConversion">End of Conversion pin</param>
+        /// <param name="controller">The GPIO controller for defined external pins. If not specified, the default controller will be used.</param>
+        /// <param name="pinNumberingScheme">Pin Numbering Scheme</param>
+        /// <param name="shouldDispose">True to dispose the Gpio Controller</param>
+        public Tlc1543(int address, int chipSelect, int dataOut, int inputOutputClock, int endOfConversion = -1, GpioController controller = null, PinNumberingScheme pinNumberingScheme = PinNumberingScheme.Logical, bool shouldDispose = true)

```

Comment:
this should take SpiDevice as input, so constructor should look like following:

```csharp
public Tlc1543(SpiDevice spiDevice)
```

endOfConversionPin - since you're not using it anywhere just do not require it since it doesn't matter anyway.

Reply:
note: address will be MOSI pin, dataOut will be MISO pin, inputOutputClock will be CLK